### ISSUE TITLE
Failed to parse a file with extension "par"

### ISSUE THREAD

**Issue by javierfreire** [aquasecurity/trivy#4040]
## Description

When trivy found a file with the extension `par`, trivy tries to unzip it. If this file is not a zip file then it fails and stops the scanning.

## What did you expect to happen?

I expect trivy to ignore this file and continue the scan.

## What happened instead?

It fails and stops the scanning.

## Output of run with `-debug`:

```
2023-04-12T11:37:12.305+0200    DEBUG   ["cyclonedx" "spdx" "spdx-json" "github"] automatically enables '--list-all-pkgs'.
2023-04-12T11:37:12.305+0200    DEBUG   Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-04-12T11:37:12.306+0200    INFO    "--format spdx" and "--format spdx-json" disable security scanning
2023-04-12T11:37:12.312+0200    DEBUG   cache dir:  /home/fjavier/.cache/trivy
2023-04-12T11:37:12.313+0200    DEBUG   Walk the file tree rooted at '.' in parallel
2023-04-12T11:37:12.313+0200    INFO    JAR files found
2023-04-12T11:37:12.313+0200    INFO    Analyzing JAR files takes a while...
2023-04-12T11:37:12.313+0200    DEBUG   Parsing Java artifacts...       {"file": "t1_blackhole.par"}
2023-04-12T11:37:12.314+0200    FATAL   rootfs scan error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.Run
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:435
  - scan error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.(*runner).scanArtifact
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:266
  - scan failed:
    github.com/aquasecurity/trivy/pkg/commands/artifact.scan
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:679
  - failed analysis:
    github.com/aquasecurity/trivy/pkg/scanner.Scanner.ScanArtifact
        /home/runner/work/trivy/trivy/pkg/scanner/scan.go:146
  - post analysis error:
    github.com/aquasecurity/trivy/pkg/fanal/artifact/local.Artifact.Inspect
        /home/runner/work/trivy/trivy/pkg/fanal/artifact/local/fs.go:164
  - post analysis error:
    github.com/aquasecurity/trivy/pkg/fanal/analyzer.AnalyzerGroup.PostAnalyze
        /home/runner/work/trivy/trivy/pkg/fanal/analyzer/analyzer.go:487
  - walk dir error:
    github.com/aquasecurity/trivy/pkg/fanal/analyzer/language/java/jar.(*javaLibraryAnalyzer).PostAnalyze
        /home/runner/work/trivy/trivy/pkg/fanal/analyzer/language/java/jar/jar.go:86
  - on file:
    github.com/aquasecurity/trivy/pkg/parallel.walk[...]
        /home/runner/work/trivy/trivy/pkg/parallel/walk.go:107
  - failed to parse t1_blackhole.par:
    github.com/aquasecurity/trivy/pkg/fanal/analyzer/language.ParsePackage
        /home/runner/work/trivy/trivy/pkg/fanal/analyzer/language/analyze.go:62
  - zip error:
    github.com/aquasecurity/go-dep-parser/pkg/java/jar.(*Parser).parseArtifact
        /home/runner/go/pkg/mod/github.com/aquasecurity/go-dep-parser@v0.0.0-20230324043952-2172dc218241/pkg/java/jar/parse.go:82
  - zip: not a valid zip file
```

## Output of `trivy -v`:

```
Version: 0.39.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-04-12 06:08:58.279551471 +0000 UTC
  NextUpdate: 2023-04-12 12:08:58.279550971 +0000 UTC
  DownloadedAt: 2023-04-12 07:35:26.388976833 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-04-12 00:51:07.355031613 +0000 UTC
  NextUpdate: 2023-04-15 00:51:07.355031213 +0000 UTC
  DownloadedAt: 2023-04-12 07:41:36.176527897 +0000 UTC
```

**Comment by DmitriyLewen** [comment]
Hello @javierfreire 
Thanks for your report.

I close this issue because it is duplicate of #3869.
We already have #4037 to fix this problem.

Regards, Dmitriy
